{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gongwei's RNASeq Data Analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Step 1:\n",
    "Create working directory in O2 server. <br>\n",
    "``` mkdir /n/scratch2/ajit ```\n",
    "#### Step 2:\n",
    "Clone BcBio pipeline into the directory. <br>\n",
    "This clones the data into a new directory named *rnaseq* <br>\n",
    "``` git clone https://github.com/datarail/rnaseq.git  ```\n",
    "``` cd rnaseq   ```\n",
    "#### Step 3:\n",
    "Create directory to download fastq files and copy the data into it. <br>\n",
    "``` mkdir /n/scratch2/ajit/rnaseq/allsamples  ``` <br>\n",
    "``` scp -r \"/mnt/c/Users/ajn16/Downloads/gw/\" ajn16@transfer.rc.hms.harvard.edu:/n/scratch2/ajit/rnaseq/allsamples/ ```<br>\n",
    "``` cd allsamples   ```\n",
    "#### Step 4:\n",
    "Create a sample description file and copy it into the *allsamples* folder. This will serve as the template for merging the same samples from different lanes into one *fastq* file. <br>\n",
    "```scp -r \"/mnt/c/Users/ajn16/Dropbox (Partners HealthCare)/Jupyter Notebooks/runs/gw_rnaseq_metadata/merge_config.csv\" ajn16@transfer.rc.hms.harvard.edu:/n/scratch2/ajit/rnaseq/allsamples/ ```<br>\n",
    "<br>\n",
    "```scp -r \"/mnt/c/Users/ajn16/Dropbox (Partners HealthCare)/Jupyter Notebooks/runs/gw_rnaseq_metadata/doMerge.sh\" ajn16@transfer.rc.hms.harvard.edu:/n/scratch2/ajit/rnaseq/allsamples/``` <br>\n",
    "<br>\n",
    "In-order to merge the files we also need a *doMerge.sh* script. Copy this also into the same folder as above. <br>\n",
    "**Sample description file**: https://github.com/ajitjohnson/Jupyter-Notebooks/blob/master/runs/gw_rnaseq_metadata/merge_config.csv <br>\n",
    "**doMerge.sh file**: https://github.com/ajitjohnson/Jupyter-Notebooks/blob/master/runs/gw_rnaseq_metadata/doMerge.sh <br>\n",
    "#### Step 5:\n",
    "Run doMerge.sh to combine the four lanes into a single fastq file (for each sample) <br>\n",
    "``` ./doMerge.sh  ```\n",
    "#### Step 6:\n",
    "Move the merged files to the *rnaseq* directory <br>\n",
    "``` mv -v /n/scratch2/ajit/rnaseq/allsamples/merged /n/scratch2/ajit/rnaseq/   ```\n",
    "<br>\n",
    "Move the **merge_config-merged.csv** file to *rnaseq* directory and rename it **alignment.csv**<br>\n",
    "``` mv /n/scratch2/ajit/rnaseq/allsamples/merge_config-merged.csv /n/scratch2/ajit/rnaseq/  ```<br>\n",
    "```cd .. ``` <br>\n",
    "```mv merge_config-merged.csv alignment.csv ```\n",
    "#### Step 7:\n",
    "Create a directory for downloading the reference genome <br>\n",
    "``` mkdir reference ```<br>\n",
    "``` cd reference/ ``` <br>\n",
    "Download reference genome <br>\n",
    "```\n",
    "wget ftp://ftp.ensembl.org/pub/release-95/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh38.cdna.all.fa.gz\n",
    "wget ftp://ftp.ensembl.org/pub/release-95/gtf/homo_sapiens/Homo_sapiens.GRCh38.95.gtf.gz\n",
    "wget ftp://ftp.ensembl.org/pub/release-95/fasta/homo_sapiens/ncrna/Homo_sapiens.GRCh38.ncrna.fa.gz\n",
    "```\n",
    "#### Step 8:\n",
    "UnZip all files \n",
    "```gunzip *.gz```\n",
    "#### Step 9:\n",
    "Concatenate cdna and ncrna -> Homo_sapiens.GRCh38.95.fa <br>\n",
    "```cat Homo_sapiens.GRCh38.cdna.all.fa Homo_sapiens.GRCh38.ncrna.fa > ./Homo_sapiens.GRCh38.95.fa ```\n",
    "#### Step 10:\n",
    "Edit **alignment.csv** if necessary: Remove unwanter ,'s <br>\n",
    "``` vim /n/scratch2/ajit/rnaseq/alignment.csv```\n",
    "#### Step 11:\n",
    "Edit **yaml_O2.yaml** if necessary <br>\n",
    "``` vim /n/scratch2/ajit/rnaseq/yaml_O2.yaml```\n",
    "Example yamal file: https://github.com/ajitjohnson/Jupyter-Notebooks/blob/master/runs/gw_rnaseq_metadata/yaml_O2.yaml <br>\n",
    "#### Step 12:\n",
    "Initiate the bcbio workspace <br>\n",
    "``` bcbio_nextgen.py -w template yaml_O2.yaml alignment.csv merged/```\n",
    "#### Step 13:\n",
    "Descend into the work subdirectory of your project <br>\n",
    "``` cd /n/scratch2/ajit/rnaseq/alignment/work ```\n",
    "#### Step 14:\n",
    "Edit *submit_bcbio.sh* if necessary <br>\n",
    "```vim /n/scratch2/ajit/rnaseq/submit_bcbio.sh```<br>\n",
    "and add <br>\n",
    "#SBATCH --mail-type=ALL <br>\n",
    "#SBATCH --mail-user=ajit_nirmal@hms.harvard.edu <br>\n",
    "<br>\n",
    "From the work directory, submit the job to O2 using the following command: <br>\n",
    "``` sbatch ../../submit_bcbio.sh ```\n",
    "#### Step 15:\n",
    "Check job progress <br>\n",
    "```sacct```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
