{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**01.14.2020** <br>\n",
    "**RNASeq Processing (fq files to counts table)** <br>\n",
    "All files were transferred to local computing cluster O2 <br>\n",
    "The cluster location is **/n/scratch2/ajit**- So replace this with your own location for reuse of this code. <br>\n",
    "\n",
    "All files were stored inside a folder named **jain_rnaseq** <br>\n",
    "Full location: /n/scratch2/ajit/jain_rnaseq/raw_files <br>\n",
    "\n",
    "**Step-1: Unzip all the files**\n",
    "\n",
    "```\n",
    "cd /n/scratch2/ajit/jain_rnaseq/raw_files\n",
    "find . -name \"*.gz\" | while read filename; do gunzip \"`dirname \"$filename\"`\" \"$filename\"; done;\n",
    "```\n",
    "\n",
    "**Step-2: Move all the files fom inside a folder to outside and remove empty folders**\n",
    "\n",
    "```\n",
    "find . -name '*.fq' -exec mv {} . \\;\n",
    "find . -depth -type d -empty -exec rmdir {} \\;\n",
    "```\n",
    "\n",
    "**Step-3: Create a sample description file and edit to include metadata regarding the samples**\n",
    "\n",
    "```\n",
    "(echo 'samplename,description'; for f in raw_files/*fq*; do readlink -f $f | perl -pe 's/(.*?_(S[0-9]+)_.*)/\\1,\\2/'; done) > alignment.csv\n",
    "```\n",
    "\n",
    "**Step-4: Download the reference genome**\n",
    "\n",
    "```\n",
    "mkdir reference\n",
    "cd reference\n",
    "wget ftp://ftp.ensembl.org/pub/release-96/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh38.cdna.all.fa.gz\n",
    "wget ftp://ftp.ensembl.org/pub/release-96/gtf/homo_sapiens/Homo_sapiens.GRCh38.96.gtf.gz\n",
    "gunzip *\n",
    "```\n",
    "\n",
    "**Step-5: Prepare the O2.yaml file**\n",
    "\n",
    "```\n",
    "cd ..\n",
    "vim O2.yaml\n",
    "\n",
    "details:\n",
    "  - analysis: RNA-seq\n",
    "    genome_build: hg38\n",
    "    algorithm:\n",
    "      transcriptome_fasta: /n/scratch2/ajit/jain_rnaseq/reference/Homo_sapiens.GRCh38.cdna.all.fa\n",
    "      transcriptome_gtf: //n/scratch2/ajit/jain_rnaseq/reference/Homo_sapiens.GRCh38.96.gtf\n",
    "      aligner: hisat2\n",
    "      strandedness: unstranded\n",
    "      tools_on: [bcbiornaseq]\n",
    "      bcbiornaseq:\n",
    "          organism: homo sapiens\n",
    "upload:\n",
    "  dir: ../final\n",
    "```\n",
    "\n",
    "**Step-6: Intiate bcBio**\n",
    "\n",
    "```\n",
    "module load bcbio/latest\n",
    "unset PYTHONPATH\n",
    "bcbio_nextgen.py -w template O2.yaml alignment.csv raw_files/\n",
    "```\n",
    "\n",
    "**Step-7: Create Submission script to O2**\n",
    "\n",
    "```\n",
    "vim submit_bcbio.sh\n",
    "\n",
    "#!/bin/sh\n",
    "#SBATCH -p medium\n",
    "#SBATCH -J bcbio_O2_jain\n",
    "#SBATCH -o run.o\n",
    "#SBATCH -e run.e\n",
    "#SBATCH -t 3-00:00\n",
    "#SBATCH --cpus-per-task=3\n",
    "#SBATCH --mem=100G\n",
    "#SBATCH --mail-type=END         # Type of email notification- BEGIN,END,FAIL,ALL\n",
    "#SBATCH --mail-user=ajitj_nirmal@dfci.harvard.edu   # Email to which notifications will be sent\n",
    "\n",
    "export PATH=/n/app/bcbio/tools/bin:$PATH\n",
    "bcbio_nextgen.py ../config/alignment.yaml \\\n",
    "    -n 24 -t ipython -s slurm -q medium -r t=3-00:00 --timeout 2000\n",
    "```\n",
    "\n",
    "**Step-8: Submit job to O2 for processing**\n",
    "\n",
    "```\n",
    "cp submit_bcbio.sh alignment/work\n",
    "cd alignment/work\n",
    "sbatch submit_bcbio.sh\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
