{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Inorder to merge samples from different lanes Create a sample description file and annotate it**\n",
    "\n",
    "```\n",
    "(echo 'samplename,description'; for f in fastq*; do readlink -f $f | perl -pe 's/(.*?_(S[0-9]+)_.*)/\\1,\\2/'; done) > tomerge.csv \n",
    "```\n",
    "\n",
    "**Submit a job to merge the files togeather**\n",
    "\n",
    "```\n",
    "#!/bin/sh\n",
    "#SBATCH -p medium\n",
    "#SBATCH -J bcbio_merge\n",
    "#SBATCH -o run.o\n",
    "#SBATCH -e run.e\n",
    "#SBATCH -t 1-00:00\n",
    "#SBATCH --cpus-per-task=3\n",
    "#SBATCH --mem=80G\n",
    "#SBATCH --mail-type=END         # Type of email notification- BEGIN,END,FAIL,ALL\n",
    "#SBATCH --mail-user=ajitj_nirmal@dfci.harvard.edu   # Email to which notifications will be sent\n",
    "\n",
    "module load bcbio/latest\n",
    "bcbio_prepare_samples.py --out merged --csv tomerge.csv\n",
    "```\n",
    "\n",
    "**Prepare alignment.csv**\n",
    "The tomerge-merged.csv produced by bcbio was edited to include meta data and alignment.csv was prepared.\n",
    "\n",
    "**Download the reference genome**\n",
    "\n",
    "```\n",
    "mkdir reference\n",
    "wget ftp://ftp.ensembl.org/pub/release-96/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh38.cdna.all.fa.gz\n",
    "wget ftp://ftp.ensembl.org/pub/release-96/gtf/homo_sapiens/Homo_sapiens.GRCh38.96.gtf.gz\n",
    "gunzip *\n",
    "```\n",
    "\n",
    "**Prepare the O2.yaml file**\n",
    "```\n",
    "details:\n",
    "  - analysis: RNA-seq\n",
    "    genome_build: hg38\n",
    "    algorithm:\n",
    "      transcriptome_fasta: /n/scratch2/ajit/FC_05641/reference/Homo_sapiens.GRCh38.cdna.all.fa\n",
    "      transcriptome_gtf: /n/scratch2/ajit/FC_05641/reference/Homo_sapiens.GRCh38.96.gtf\n",
    "      aligner: hisat2\n",
    "      trim_reads: read_through\n",
    "      adapters: [nextera, polya]\n",
    "      strandedness: unstranded\n",
    "      tools_on: [bcbiornaseq]\n",
    "      bcbiornaseq:\n",
    "          organism: homo sapiens\n",
    "          interesting_groups: region\n",
    "upload:\n",
    "  dir: ../final\n",
    "```\n",
    "\n",
    "**Intiate bcBio**\n",
    "\n",
    "```\n",
    "module load bcbio/latest\n",
    "unset PYTHONPATH\n",
    "bcbio_nextgen.py -w template O2.yaml alignment.csv raw_files/\n",
    "```\n",
    "\n",
    "**Submit Job to O2**\n",
    "\n",
    "```\n",
    "cd alignment/work\n",
    "vim submit_bcbio.sh\n",
    "\n",
    "#!/bin/sh\n",
    "#SBATCH -p medium\n",
    "#SBATCH -J bcbio_O2              \n",
    "#SBATCH -o run.o\n",
    "#SBATCH -e run.e\n",
    "#SBATCH -t 5-00:00\n",
    "#SBATCH --cpus-per-task=3\n",
    "#SBATCH --mem=80G\n",
    "#SBATCH --mail-type=END         # Type of email notification- BEGIN,END,FAIL,ALL\n",
    "#SBATCH --mail-user=ajitj_nirmal@dfci.harvard.edu   # Email to which notifications will be sent\n",
    "\n",
    "export PATH=/n/app/bcbio/tools/bin:$PATH\n",
    "bcbio_nextgen.py ../config/alignment.yaml \\\n",
    "    -n 24 -t ipython -s slurm -q medium -r t=5-00:00 --timeout 2000\n",
    "\n",
    "sbatch submit_bcbio.sh\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
