{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Step 1: Rename files and make sure every chip file has an associated input file\n",
    "All the data was manually reviewed and renamed for human readability. Since there was a single input sample seqenced that would serve as an input for several chipseq files, the input was duplicated such that each chipseq file has an associated input file. The data was then transfered to O2 for processing. <br>\n",
    "\n",
    "```\n",
    "srun --pty -p interactive --mem 80G -t 0-06:00 /bin/bash\n",
    "gunzip *.gz\n",
    "mkdir reference\n",
    "wget ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/hg19.zip\n",
    "unzip hg19.zip\n",
    "```\n",
    "\n",
    "#### Step 2: \n",
    "\n",
    "```\n",
    "vim run_1.sh\n",
    "```\n",
    "\n",
    "```\n",
    "#!/bin/sh\n",
    "#SBATCH -p short\n",
    "#SBATCH -J bowtie2_O2\n",
    "#SBATCH -o run.o\n",
    "#SBATCH -e run.e\n",
    "#SBATCH -t 0-12:00                                                                                                    #SBATCH --cpus-per-task=1\n",
    "#SBATCH --mem=8000\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH --mail-user=ajit_nirmal@hms.harvard.edu\n",
    "\n",
    "module load gcc/6.2.0 bowtie2/2.3.4.3\n",
    "\n",
    "/n/app/bowtie2/2.3.4.3/bin/bowtie2 -x hg19 -1 /n/scratch2/ajit/gw_chip/fastq/GW_K27Ac_2_FRRC190725836-1b_H3KNFBBXX_L7_1.fq.gz -2 /n/scratch2/ajit/gw_chip/fastq/GW_K27Ac_2_FRRC190725836-1b_H3KNFBBXX_L7_2.fq.gz -S GW_K27Ac_2.sam\n",
    "```\n",
    "\n",
    "#### Step 3:\n",
    "The above generated SAM files were converted into BAM files and sorted using samtools. <br>\n",
    "\n",
    "```\n",
    "#!/bin/sh\n",
    "#SBATCH -p short\n",
    "#SBATCH -J bowtie2_O2_samtobam\n",
    "#SBATCH -o run.osamtobam\n",
    "#SBATCH -e run.esamtobam\n",
    "#SBATCH -t 0-4:00\n",
    "#SBATCH --cpus-per-task=1\n",
    "#SBATCH --mem=8000\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH --mail-user=ajit_nirmal@hms.harvard.edu\n",
    "\n",
    "module load gcc/6.2.0 samtools/1.9\n",
    "\n",
    "samtools view -S -b GW_K27Ac_2.sam > GW_K27Ac_2.bam\n",
    "samtools sort GW_K27Ac_2.bam -o sorted_GW_K27Ac_2.bam\n",
    "```\n",
    "\n",
    "##### Using BAM files: MACS will use the real fragments inferred from alignment results for reads pileup.\n",
    "\n",
    "```\n",
    "#!/bin/sh\n",
    "#SBATCH -p short\n",
    "#SBATCH -J bowtie2_O2_J\n",
    "#SBATCH -o run.om\n",
    "#SBATCH -e run.em\n",
    "#SBATCH -t 0-03:00\n",
    "#SBATCH --cpus-per-task=1\n",
    "#SBATCH --mem=8000\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH --mail-user=ajit_nirmal@hms.harvard.edu\n",
    "\n",
    "module load gcc/6.2.0 python/2.7.12 macs2/2.1.1.20160309\n",
    "\n",
    "macs2 callpeak -t sorted_GW_TP63.bam -f BAMPE -g hs -n JPE -B -q 0.05\n",
    "```\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
