{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Exome analysis of Patient 16042-023 treated with Duv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Move all the files fom inside a folder to outside\n",
    "\n",
    "```\n",
    "mkdir to_merge\n",
    "find . -name '*.fq.gz' -exec mv {} . \\;\n",
    "mv *.fq.gz to_merge/\n",
    "find . -name '*.txt' -exec mv {} . \\;\n",
    "find . -depth -type d -empty -exec rmdir {} \\;\n",
    "```\n",
    "\n",
    "##### Block to merge multiple lanes into one\n",
    "```\n",
    "# Unzip all files\n",
    "gunzip *\n",
    "\n",
    "# Create to_merge csv file\n",
    "(echo 'samplename,description'; for f in to_merge/*fq; do readlink -f $f | perl -pe 's/(.*?_(S[0-9]+)_.*)/\\1,\\2/'; done) > to_merge.csv\n",
    "\n",
    "vim merge.sh\n",
    "\n",
    "#!/bin/sh\n",
    "#SBATCH -p priority\n",
    "#SBATCH -J merge\n",
    "#SBATCH -o run.o\n",
    "#SBATCH -e run.e\n",
    "#SBATCH -t 0-12:00\n",
    "#SBATCH --cpus-per-task=20\n",
    "#SBATCH --mail-type=END         # Type of email notification- BEGIN,END,FAIL,ALL\n",
    "#SBATCH --mail-user=ajitj_nirmal@dfci.harvard.edu   # Email to which notifications will be sent\n",
    "\n",
    "module load bcbio/latest\n",
    "cd to_merge\n",
    "bcbio_prepare_samples.py --out merged --csv to_merge.csv\n",
    "\n",
    "sbatch merge.sh \n",
    "\n",
    "```\n",
    "\n",
    "##### Make a copy of the vechile samples and rename them\n",
    "\n",
    "```\n",
    "# Three comparsions\n",
    "# 70- pre, 76- post, 77- germline\n",
    "# 1. Post-treatment vs Pre-treatment (76 vs 70)\n",
    "# 2. Post-treatment vs germline (76 vs 77)\n",
    "# 3. Pre-treatment vs germline (70 vs 77)\n",
    "\n",
    "mkdir c1 c2 c3\n",
    "for dir in */; do mkdir -- \"$dir/raw\"; done\n",
    "```\n",
    "\n",
    "##### Create Alignment file\n",
    "\n",
    "```\n",
    "(echo 'samplename,description'; for f in raw/*.gz; do readlink -f $f | perl -pe 's/(.*?(S[0-9]+)_.*)/\\1,\\2/'; done) > alignment.csv\n",
    "```\n",
    "\n",
    "##### Yaml file\n",
    "\n",
    "```\n",
    "vim O2.yaml\n",
    "\n",
    "details:\n",
    "- analysis: variant2\n",
    "  genome_build: hg38\n",
    "  algorithm:\n",
    "    aligner: bwa\n",
    "    tools_on: [gemini]\n",
    "    remove_lcr: true\n",
    "    variantcaller: [mutect2]\n",
    "    variant_regions: /n/scratch3/users/a/ajn16/exome/run/input/S07604514_Regions.bed\n",
    "    svcaller: [cnvkit, lumpy, delly]\n",
    "upload:\n",
    "  dir: ../final\n",
    "  \n",
    "```\n",
    "\n",
    "##### Sbatch script\n",
    "\n",
    "```\n",
    "#!/bin/sh\n",
    "#SBATCH -p priority\n",
    "#SBATCH -J exome_1\n",
    "#SBATCH -o run.o\n",
    "#SBATCH -e run.e\n",
    "#SBATCH -t 2-00:00\n",
    "#SBATCH --cpus-per-task=20\n",
    "#SBATCH --mem=80G\n",
    "#SBATCH --mail-type=END         # Type of email notification- BEGIN,END,FAIL,ALL\n",
    "#SBATCH --mail-user=ajitj_nirmal@dfci.harvard.edu   # Email to which notifications will be sent\n",
    "\n",
    "export PATH=/n/app/bcbio/tools/bin:$PATH\n",
    "bcbio_nextgen.py ../config/alignment.yaml \\\n",
    "    -n 24 -t local\n",
    "```\n",
    "\n",
    "##### Run bcbio\n",
    "\n",
    "```\n",
    "module load bcbio/latest\n",
    "unset PYTHONPATH\n",
    "bcbio_nextgen.py -w template O2.yaml alignment.csv raw/ --separator '_'\n",
    "```\n",
    "\n",
    "##### Submit Job\n",
    "\n",
    "```\n",
    "cp submit_sbatch.sh alignment/work/\n",
    "cd alignment/work\n",
    "sbatch submit_sbatch.sh\n",
    "```\n",
    "\n",
    "##### Working with the VCF files\n",
    "\n",
    "```\n",
    "# Merge all vcf entries\n",
    "module load gcc/6.2.0 bcftools/1.10.2\n",
    "\n",
    "# did not work\n",
    "bgzip c1.vcf\n",
    "bgzip c2.vcf\n",
    "bgzip c3.vcf\n",
    "bcftools index c1.vcf.gz && bcftools index c2.vcf.gz && bcftools index c3.vcf.gz\n",
    "bcftools merge --merge all c1.vcf.gz c2.vcf.gz c3.vcf.gz > merged.vcf\n",
    "\n",
    "# convert vcf to maf and merge later\n",
    "conda activate vcf2maf\n",
    "\n",
    "# remove line breaks\n",
    "perl -lne 's/\\r//; print \"$_\";' c1.vcf > c1_clean.vcf\n",
    "perl -lne 's/\\r//; print \"$_\";' c2.vcf > c2_clean.vcf\n",
    "perl -lne 's/\\r//; print \"$_\";' c3.vcf > c3_clean.vcf\n",
    "\n",
    "# copy the vcf2maf.pl\n",
    "cp ~/softwares/vcf2maf/mskcc-vcf2maf-47c4a18/vcf2maf.pl .\n",
    "\n",
    "# Run the vcf2maf tool\n",
    "perl vcf2maf.pl --input-vcf /n/scratch3/users/a/ajn16/VCF/vcf2/c1_clean.vcf --output-maf maf/c1.maf --inhibit-vep --ref-fasta /n/shared_db/bcbio/biodata/genomes/Hsapiens/hg38/seq/hg38.fa.gz --filter-vcf 0\n",
    "\n",
    "perl vcf2maf.pl --input-vcf /n/scratch3/users/a/ajn16/VCF/vcf2/c2_clean.vcf --output-maf maf/c2.maf --inhibit-vep --ref-fasta /n/shared_db/bcbio/biodata/genomes/Hsapiens/hg38/seq/hg38.fa.gz --filter-vcf 0\n",
    "\n",
    "perl vcf2maf.pl --input-vcf /n/scratch3/users/a/ajn16/VCF/vcf2/c3_clean.vcf --output-maf maf/c3.maf --inhibit-vep --ref-fasta /n/shared_db/bcbio/biodata/genomes/Hsapiens/hg38/seq/hg38.fa.gz --filter-vcf 0\n",
    "\n",
    "```\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
