{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**UNZIP data**\n",
    "```\n",
    "soar\n",
    "cd /n/scratch2/ajit/duv_pdx/rawdata\n",
    "find . -name \"*.gz\" | while read filename; do gunzip \"`dirname \"$filename\"`\" \"$filename\"; done;\n",
    "```\n",
    "**Create a sample description file**\n",
    "```\n",
    "alignment.csv\n",
    "```\n",
    "\n",
    "**Download Reference genome**\n",
    "\n",
    "```\n",
    "mkdir reference\n",
    "wget ftp://ftp.ensembl.org/pub/release-96/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh38.cdna.all.fa.gz\n",
    "wget ftp://ftp.ensembl.org/pub/release-96/gtf/homo_sapiens/Homo_sapiens.GRCh38.96.gtf.gz\n",
    "gunzip *\n",
    "```\n",
    "\n",
    "**Create YAML File**\n",
    "\n",
    "```\n",
    "vim o2.yaml\n",
    "details:\n",
    "  - analysis: RNA-seq\n",
    "    genome_build: hg38\n",
    "    algorithm:\n",
    "      transcriptome_fasta: /n/scratch2/ajit/duv_pdx/reference/Homo_sapiens.GRCh38.cdna.all.fa\n",
    "      transcriptome_gtf: /n/scratch2/ajit/duv_pdx/reference/Homo_sapiens.GRCh38.96.gtf\n",
    "      aligner: hisat2\n",
    "      strandedness: unstranded\n",
    "      tools_on: [bcbiornaseq]\n",
    "      bcbiornaseq:\n",
    "          organism: homo sapiens\n",
    "          interesting_groups: conditions\n",
    "upload:\n",
    "  dir: ../final\n",
    "```\n",
    "\n",
    "**Intiate bcBio**\n",
    "\n",
    "```bash\n",
    "module load bcbio/latest\n",
    "unset PYTHONPATH\n",
    "bcbio_nextgen.py -w template o2.yaml alignment.csv rawdata/ --separator '_'\n",
    "```\n",
    "\n",
    "**Run bcBio**\n",
    "\n",
    "```bash\n",
    "cd alignment/work\n",
    "vim submit_bcbio.sh\n",
    "\n",
    "#!/bin/sh\n",
    "#SBATCH -p medium\n",
    "#SBATCH -J bcbio_O2              \n",
    "#SBATCH -o run.o\n",
    "#SBATCH -e run.e\n",
    "#SBATCH -t 1-00:00\n",
    "#SBATCH --cpus-per-task=3\n",
    "#SBATCH --mem=100G\n",
    "#SBATCH --mail-type=END         # Type of email notification- BEGIN,END,FAIL,ALL\n",
    "#SBATCH --mail-user=ajitj_nirmal@dfci.harvard.edu   # Email to which notifications will be sent\n",
    "\n",
    "export PATH=/n/app/bcbio/tools/bin:$PATH\n",
    "bcbio_nextgen.py ../config/alignment.yaml \\\n",
    "    -n 24 -t ipython -s slurm -q medium -r t=1-00:00 --timeout 2000\n",
    "\n",
    "sbatch submit_bcbio.sh\n",
    "```\n",
    "\n",
    "---\n",
    "\n",
    "**Processing EXOME sequecing data**\n",
    "\n",
    "```\n",
    "srun --pty -p interactive --mem 70G -t 0-06:00 /bin/bash\n",
    "find . -name \"*.gz\" | while read filename; do gunzip \"`dirname \"$filename\"`\" \"$filename\"; done;\n",
    "```\n",
    "\n",
    "**Move all the files fom inside a folder to outside**\n",
    "\n",
    "```\n",
    "find . -name '*.fq' -exec mv {} . \\;\n",
    "find . -depth -type d -empty -exec rmdir {} \\;\n",
    "```\n",
    "\n",
    "**Move extra smaller files into a directory**\n",
    "\n",
    "```\n",
    "du -bsh *\n",
    "```\n",
    "\n",
    "**Make a copy of the vechile samples and renmae them**\n",
    "\n",
    "```\n",
    "mkdir vechile_1\n",
    "cp V* vechile_1\n",
    "mkdir vechile_2\n",
    "cp V* vechile_2/\n",
    "\n",
    "cd vechile_1\n",
    "rename 'USD16092608' comb *\n",
    "rename 'USD16092609' comb *\n",
    "rename 'USD16092610' comb *\n",
    "\n",
    "cd vechile_2\n",
    "rename 'USD16092608' romi *\n",
    "rename 'USD16092609' romi *\n",
    "rename 'USD16092610' romi *\n",
    "```\n",
    "\n",
    "**Rename the files**\n",
    "\n",
    "```\n",
    "for file in *; do mv \"${file}\" \"${file/_HMGWWDSXX_L4/}\"; done # Remove defined string from name\n",
    "for file in *; do mv \"${file}\" \"${file/s47880_/}\"; done # Remove defined string from name\n",
    "```\n",
    "\n",
    "**Create a sample description file**\n",
    "\n",
    "\n",
    "(echo 'samplename,description'; for f in raw_files/*fq*; do readlink -f $f | perl -pe 's/(.*?_(S[0-9]+)_.*)/\\1,\\2/'; done) > alignment.csv\n",
    "\n",
    "Edit alignment.csv remotely to include phenotype, batch and drug details.\n",
    "\n",
    "**Create the YAML file**\n",
    "\n",
    "```\n",
    "details:\n",
    "- analysis: variant2\n",
    "  genome_build: hg38\n",
    "  algorithm:\n",
    "    aligner: bwa\n",
    "    disambiguate: [mm10]\n",
    "    tools_on: [gemini]\n",
    "    remove_lcr: true\n",
    "    variantcaller: [mutect2]\n",
    "    variant_regions: ../input/S07604514_Regions.bed\n",
    "    svcaller: [cnvkit, lumpy, delly]\n",
    "upload:\n",
    "  dir: ../final\n",
    "```\n",
    "\n",
    "**Download the variant region file**\n",
    "\n",
    "Download the BED files from https://earray.chem.agilent.com/suredesign/index.htm and save it inside a folder named input\n",
    "\n",
    "**Create the submission script**\n",
    "\n",
    "```\n",
    "#!/bin/sh\n",
    "#SBATCH -p priority\n",
    "#SBATCH -J exome\n",
    "#SBATCH -o run.o\n",
    "#SBATCH -e run.e\n",
    "#SBATCH -t 7-00:00\n",
    "#SBATCH --cpus-per-task=20\n",
    "#SBATCH --mem=80G\n",
    "#SBATCH --mail-type=END         # Type of email notification- BEGIN,END,FAIL,ALL\n",
    "#SBATCH --mail-user=ajitj_nirmal@dfci.harvard.edu   # Email to which notifications will be sent\n",
    "\n",
    "export PATH=/n/app/bcbio/tools/bin:$PATH\n",
    "bcbio_nextgen.py ../config/alignment.yaml \\\n",
    "    -n 24 -t local\n",
    "```\n",
    "\n",
    "**Run bcbio**\n",
    "\n",
    "```bash\n",
    "module load bcbio/latest\n",
    "unset PYTHONPATH\n",
    "bcbio_nextgen.py -w template O2.yaml alignment.csv raw_files/ --separator '_'\n",
    "```\n",
    "\n",
    "**Submit Job**\n",
    "\n",
    "```\n",
    "cp submit_bcbio.sh alignment/work/\n",
    "cd alignment/work\n",
    "sbatch submit_bcbio.sh\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Re-running Exome sequencing data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Merge files: multiple lanes and all vechile samples into a single entity**\n",
    "\n",
    "**Create tomerge file**\n",
    "\n",
    "```\n",
    "(echo 'samplename,description'; for f in raw_files/*; do readlink -f $f | perl -pe 's/(.?(S[0-9]+).)/\\1,\\2/'; done) > tomerge.csv\n",
    "```\n",
    "\n",
    "**Merge multiple files into a single file**\n",
    "\n",
    "```\n",
    "#!/bin/sh\n",
    "#SBATCH -p priority\n",
    "#SBATCH -J merge\n",
    "#SBATCH -o run.o\n",
    "#SBATCH -e run.e\n",
    "#SBATCH -t 0-12:00\n",
    "#SBATCH --cpus-per-task=20\n",
    "#SBATCH --mail-type=END         # Type of email notification- BEGIN,END,FAIL,ALL\n",
    "#SBATCH --mail-user=ajitj_nirmal@dfci.harvard.edu   # Email to which notifications will be sent\n",
    "\n",
    "module load bcbio/latest\n",
    "cd raw_files\n",
    "bcbio_prepare_samples.py --memory-per-job 2G --out merged --csv tomerge.csv\n",
    "```\n",
    "\n",
    "**Create a sample description file**\n",
    "\n",
    "(echo 'samplename,description'; for f in raw_files/*; do readlink -f $f | perl -pe 's/(.*?_(S[0-9]+)_.*)/\\1,\\2/'; done) > alignment.csv\n",
    "\n",
    "Edit alignment.csv remotely to include phenotype, batch and drug details.\n",
    "\n",
    "**Create the YAML file**\n",
    "\n",
    "```\n",
    "details:\n",
    "- analysis: variant2\n",
    "  genome_build: hg38\n",
    "  algorithm:\n",
    "    aligner: bwa\n",
    "    disambiguate: [mm10]\n",
    "    tools_on: [gemini]\n",
    "    remove_lcr: true\n",
    "    variantcaller: [mutect2]\n",
    "    variant_regions: ../input/S07604514_Regions.bed\n",
    "    svcaller: [cnvkit, lumpy, delly]\n",
    "upload:\n",
    "  dir: ../final\n",
    "```\n",
    "\n",
    "**Download the variant region file**\n",
    "\n",
    "Download the BED files from https://earray.chem.agilent.com/suredesign/index.htm and save it inside a folder named input\n",
    "\n",
    "**Create the submission script**\n",
    "\n",
    "```\n",
    "vim submit_bcbio.sh\n",
    "\n",
    "\n",
    "#!/bin/sh\n",
    "#SBATCH -p priority\n",
    "#SBATCH -J exome\n",
    "#SBATCH -o run.o\n",
    "#SBATCH -e run.e\n",
    "#SBATCH -t 7-00:00\n",
    "#SBATCH --cpus-per-task=20\n",
    "#SBATCH --mem=80G\n",
    "#SBATCH --mail-type=END         # Type of email notification- BEGIN,END,FAIL,ALL\n",
    "#SBATCH --mail-user=ajitj_nirmal@dfci.harvard.edu   # Email to which notifications will be sent\n",
    "\n",
    "export PATH=/n/app/bcbio/tools/bin:$PATH\n",
    "bcbio_nextgen.py ../config/alignment.yaml \\\n",
    "    -n 24 -t local\n",
    "```\n",
    "\n",
    "**Run bcbio**\n",
    "\n",
    "```bash\n",
    "module load bcbio/latest\n",
    "unset PYTHONPATH\n",
    "bcbio_nextgen.py -w template O2.yaml alignment.csv raw_files/ --separator '_'\n",
    "```\n",
    "\n",
    "**Submit Job**\n",
    "\n",
    "```\n",
    "cp submit_bcbio.sh alignment/work/\n",
    "cd alignment/work\n",
    "sbatch submit_bcbio.sh\n",
    "```\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Working with the VCF files\n",
    "\n",
    "```\n",
    "# Merge common entries\n",
    "module load gcc/6.2.0 bcftools/1.9\n",
    "\n",
    "# D1 and D2\n",
    "bcftools isec -p isec_output -Oz 4-mutect2-annotated.vcf.gz 5-mutect2-annotated.vcf.gz\n",
    "bcftools merge --merge all 0002.vcf.gz 0003.vcf.gz > D12merged.vcf\n",
    "\n",
    "# D2 and D3\n",
    "bcftools isec -p isec_output -Oz 5-mutect2-annotated.vcf.gz 6-mutect2-annotated.vcf.gz\n",
    "bcftools merge --merge all 0002.vcf.gz 0003.vcf.gz > D23merged.vcf\n",
    "\n",
    "# D1 and D3\n",
    "bcftools isec -p isec_output -Oz 4-mutect2-annotated.vcf.gz 6-mutect2-annotated.vcf.gz\n",
    "bcftools merge --merge all 0002.vcf.gz 0003.vcf.gz > D13merged.vcf\n",
    "\n",
    "# D1, D2 and D3\n",
    "bcftools isec -n +3 4-mutect2-annotated.vcf.gz 5-mutect2-annotated.vcf.gz 6-mutect2-annotated.vcf.gz -p D123\n",
    "\n",
    "# Variants in two or more VCF files\n",
    "bcftools isec -n +2 4-mutect2-annotated.vcf.gz 5-mutect2-annotated.vcf.gz 6-mutect2-annotated.vcf.gz -p Dtwo\n",
    "\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Exome analysis (12th Dec 2020)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Create Alignment file\n",
    "\n",
    "```\n",
    "(echo 'samplename,description'; for f in raw/*.gz; do readlink -f $f | perl -pe 's/(.*?(S[0-9]+)_.*)/\\1,\\2/'; done) > alignment.csv\n",
    "```\n",
    "\n",
    "#### Edit the alignemnt file as so:\n",
    "\n",
    "```\n",
    "samplename,description,batch,phenotype\n",
    "/n/scratch3/users/a/ajn16/pdx_exome/raw/D1_1.fq.gz,D1,tumor\n",
    "/n/scratch3/users/a/ajn16/pdx_exome/raw/D2_1.fq.gz,D2,tumor\n",
    "/n/scratch3/users/a/ajn16/pdx_exome/raw/D3_1.fq.gz,D3,tumor\n",
    "/n/scratch3/users/a/ajn16/pdx_exome/raw/V_1.fastq.gz,D1;D2;D3,normal\n",
    "```\n",
    "\n",
    "#### Yaml file\n",
    "\n",
    "```\n",
    "vim O2.yaml\n",
    "\n",
    "details:\n",
    "- analysis: variant2\n",
    "  genome_build: hg38\n",
    "  algorithm:\n",
    "    aligner: bwa\n",
    "    disambiguate: [mm10]\n",
    "    tools_on: [gemini]\n",
    "    remove_lcr: true\n",
    "    variantcaller: [mutect2]\n",
    "    variant_regions: /n/scratch3/users/a/ajn16/pdx_exome/input/S07604514_Regions.bed\n",
    "    svcaller: [cnvkit, lumpy, delly]\n",
    "upload:\n",
    "  dir: ../final\n",
    "  \n",
    "```\n",
    "\n",
    "#### Sbatch script\n",
    "\n",
    "````\n",
    "#!/bin/sh\n",
    "#SBATCH -p priority\n",
    "#SBATCH -J exome_1\n",
    "#SBATCH -o run.o\n",
    "#SBATCH -e run.e\n",
    "#SBATCH -t 2-00:00\n",
    "#SBATCH --cpus-per-task=20\n",
    "#SBATCH --mem=80G\n",
    "#SBATCH --mail-type=END         # Type of email notification- BEGIN,END,FAIL,ALL\n",
    "#SBATCH --mail-user=ajitj_nirmal@dfci.harvard.edu   # Email to which notifications will be sent\n",
    "\n",
    "export PATH=/n/app/bcbio/tools/bin:$PATH\n",
    "bcbio_nextgen.py ../config/alignment.yaml \\\n",
    "    -n 24 -t local\n",
    "```\n",
    "\n",
    "#### Run bcbio\n",
    "\n",
    "```\n",
    "module load bcbio/latest\n",
    "unset PYTHONPATH\n",
    "bcbio_nextgen.py -w template O2.yaml alignment.csv raw/*\n",
    "```\n",
    "\n",
    "#### Submit Job\n",
    "\n",
    "```\n",
    "cp submit_sbatch.sh alignment/work/\n",
    "cd alignment/work\n",
    "sbatch submit_sbatch.sh\n",
    "```\n",
    "\n",
    "# CNV analysis\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Collect PON samples**\n",
    "\n",
    "```\n",
    "mkdir normal\n",
    "cd normal\n",
    "mkdir input config\n",
    "```\n",
    "\n",
    "**Add Bed file to config and bam files to input**\n",
    "\n",
    "```\n",
    "cd config\n",
    "cp /n/scratch3/users/a/ajn16/pdx_exome/input/S07604514_Regions.bed .\n",
    "\n",
    "cd input\n",
    "cp /n/scratch3/users/a/ajn16/pdx_exome/alignment/final/V1/V1-ready.bam .\n",
    "cp /n/scratch3/users/a/ajn16/pdx_exome/alignment/final/D1/D1-ready.bam .\n",
    "mv V1-ready.bam V1.bam\n",
    "mv D1-ready.bam D1.bam\n",
    "```\n",
    "\n",
    "**Create pon.csv**\n",
    "\n",
    "```\n",
    "cd ..\n",
    "\n",
    "vim pon.csv\n",
    "\n",
    "samplename,description,svclass,batch\n",
    "V1.bam,V1,control,pon_build\n",
    "D1.bam,D1,tumor,pon_build\n",
    "\n",
    "```\n",
    "\n",
    "**Create pon_template.yaml**\n",
    "\n",
    "```\n",
    "vim pon_template.yaml\n",
    "\n",
    "details:\n",
    "  - analysis: variant2\n",
    "    genome_build: hg38\n",
    "    algorithm:\n",
    "      svcaller: [gatk-cnv]\n",
    "      variant_regions: /n/scratch3/users/a/ajn16/pdx_exome/cnv/normal/config/S07604514_Regions.bed\n",
    "\n",
    "```\n",
    "\n",
    "**Run bcbio**\n",
    "\n",
    "```\n",
    "module load bcbio/latest\n",
    "bcbio_nextgen.py -w template pon_template.yaml pon.csv input/*.bam\n",
    "```\n",
    "\n",
    "**submit a sbarch script to create PON files**\n",
    "\n",
    "```\n",
    "vim pon_submit.sh\n",
    "\n",
    "#!/bin/sh\n",
    "#SBATCH -p priority\n",
    "#SBATCH -J pon_1\n",
    "#SBATCH -o run.o\n",
    "#SBATCH -e run.e\n",
    "#SBATCH -t 1-00:00\n",
    "#SBATCH --cpus-per-task=20\n",
    "#SBATCH --mem=80G\n",
    "#SBATCH --mail-type=END         # Type of email notification- BEGIN,END,FAIL,ALL\n",
    "#SBATCH --mail-user=ajitj_nirmal@dfci.harvard.edu   # Email to which notifications will be sent\n",
    "\n",
    "export PATH=/n/app/bcbio/tools/bin:$PATH\n",
    "bcbio_nextgen.py ../config/pon.yaml \\\n",
    "    -n 24 -t local\n",
    "\n",
    "cp pon_submit.sh pon/work\n",
    "cd pon/work\n",
    "sbatch pon_submit.sh\n",
    "\n",
    "```\n",
    "\n",
    "**TUMOR**\n",
    "**Create pon_tn project structure**\n",
    "\n",
    "```\n",
    "mkdir pon_tn\n",
    "cd pon_tn\n",
    "mkdir config input\n",
    "cd config\n",
    "\n",
    "cp /n/scratch3/users/a/ajn16/pdx_exome/cnv/normal/pon/final/2020-12-17_pon/gatkcnv-pon.hdf5 .\n",
    "cp /n/scratch3/users/a/ajn16/pdx_exome/input/S07604514_Regions.bed .\n",
    "\n",
    "```\n",
    "\n",
    "**Copy the tumor and normal samples into the input folder**\n",
    "\n",
    "```\n",
    "# We have one normal and three tumor samples\n",
    "\n",
    "cp /n/scratch3/users/a/ajn16/pdx_exome/alignment/final/V1/V1-ready.bam .\n",
    "cp /n/scratch3/users/a/ajn16/pdx_exome/alignment/final/D1/D1-ready.bam .\n",
    "cp /n/scratch3/users/a/ajn16/pdx_exome/alignment/final/D2/D2-ready.bam .\n",
    "cp /n/scratch3/users/a/ajn16/pdx_exome/alignment/final/D3/D3-ready.bam .\n",
    "\n",
    "mv V1-ready.bam V1.bam\n",
    "mv D1-ready.bam D1.bam\n",
    "mv D2-ready.bam D2.bam\n",
    "mv D3-ready.bam D3.bam\n",
    "\n",
    "```\n",
    "\n",
    "**Prepare a sample sheet pon_tn.csv**\n",
    "\n",
    "```\n",
    "cd ..\n",
    "\n",
    "vim pon_tn.csv\n",
    "\n",
    "samplename,description,batch,phenotype\n",
    "V1.bam,V1,Da;Db;Dc,normal\n",
    "D1.bam,D1,Da,tumor\n",
    "D2.bam,D2,Db,tumor\n",
    "D3.bam,D3,Dc,tumor\n",
    "\n",
    "```\n",
    "\n",
    "**Premate a yaml template: pon_tn_template.yaml**\n",
    "    \n",
    "```\n",
    "\n",
    "vim pon_tn_template.yaml\n",
    "\n",
    "\n",
    "details:\n",
    "  - analysis: variant2\n",
    "    genome_build: hg38\n",
    "    algorithm:\n",
    "      svcaller: [gatk-cnv]\n",
    "      variant_regions: /n/scratch3/users/a/ajn16/pdx_exome/cnv/pon_tn/config/S07604514_Regions.bed\n",
    "      coverage_interval: regional\n",
    "      background:\n",
    "        cnv_reference:\n",
    "          gatk-cnv: /n/scratch3/users/a/ajn16/pdx_exome/cnv/pon_tn/config/gatkcnv-pon.hdf5\n",
    "\n",
    "```\n",
    "\n",
    "**configure pon_tn project**\n",
    "\n",
    "```\n",
    "module load bcbio/latest\n",
    "bcbio_nextgen.py -w template pon_tn_template.yaml pon_tn.csv input/*.bam\n",
    "\n",
    "```\n",
    "\n",
    "**Run bcbio pon_tn project**\n",
    "\n",
    "```\n",
    "vim cnv_submit.sh\n",
    "\n",
    "#!/bin/sh\n",
    "#SBATCH -p priority\n",
    "#SBATCH -J cnv_2\n",
    "#SBATCH -o run.o\n",
    "#SBATCH -e run.e\n",
    "#SBATCH -t 1-00:00\n",
    "#SBATCH --cpus-per-task=20\n",
    "#SBATCH --mem=80G\n",
    "#SBATCH --mail-type=END         # Type of email notification- BEGIN,END,FAIL,ALL\n",
    "#SBATCH --mail-user=ajitj_nirmal@dfci.harvard.edu   # Email to which notifications will be sent\n",
    "\n",
    "export PATH=/n/app/bcbio/tools/bin:$PATH\n",
    "bcbio_nextgen.py ../config/pon_tn.yaml \\\n",
    "    -n 24 -t local\n",
    "\n",
    "cp cnv_submit.sh pon_tn/work\n",
    "cd pon_tn/work\n",
    "sbatch cnv_submit.sh\n",
    "\n",
    "```\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Working with VCF files\n",
    "#### FInd the common variants across all three samples\n",
    "\n",
    "```\n",
    "bcftools isec -n +3 Da-mutect2-annotated.vcf.gz Db-mutect2-annotated.vcf.gz Dc-mutect2-annotated.vcf.gz -p D123\n",
    "# It will produce 3 files and all are identical (use any one)\n",
    "```\n",
    "\n",
    "#### Find variants that are present in two or more samples\n",
    "\n",
    "```\n",
    "bcftools isec -n +2 Da-mutect2-annotated.vcf.gz Db-mutect2-annotated.vcf.gz Dc-mutect2-annotated.vcf.gz -p Dtwo\n",
    "# It will again produce 3 files but the variants will be different. Merge the three files to identify the union\n",
    "\n",
    "bcftools view 0000.vcf -Oz -o 0000.vcf.gz\n",
    "bcftools view 0001.vcf -Oz -o 0001.vcf.gz\n",
    "bcftools view 0002.vcf -Oz -o 0002.vcf.gz\n",
    "\n",
    "# index them\n",
    "bcftools index 0000.vcf.gz\n",
    "bcftools index 0001.vcf.gz\n",
    "bcftools index 0002.vcf.gz\n",
    "\n",
    "\n",
    "# Merge (did not work) some propblem with VCF file due to GATK processing\n",
    "bcftools merge 0000.vcf.gz  0001.vcf.gz  0002.vcf.gz --force-samples -O v -o merged.vcf\n",
    "```\n",
    "\n",
    "#### Finding common CNV's\n",
    "\n",
    "```\n",
    "bcftools isec -n +3 Da-gatk-cnv.vcf.gz Db-gatk-cnv.vcf.gz Dc-gatk-cnv.vcf.gz -p D123\n",
    "bcftools isec -n +2 Da-gatk-cnv.vcf.gz Db-gatk-cnv.vcf.gz Dc-gatk-cnv.vcf.gz -p Dtwo\n",
    "\n",
    "```\n",
    "\n",
    "#### From Novogene\n",
    "\n",
    "```\n",
    "#SNP\n",
    "bcftools index s47880_D1.GATK.snp.vcf.gz \n",
    "bcftools index s47880_D2.GATK.snp.vcf.gz\n",
    "bcftools index s47880_D3.GATK.snp.vcf.gz\n",
    "bcftools isec -n +3 s47880_D1.GATK.snp.vcf.gz s47880_D2.GATK.snp.vcf.gz s47880_D3.GATK.snp.vcf.gz -p D123\n",
    "\n",
    "#INDEL\n",
    "bcftools index s47880_D1.GATK.indel.vcf.gz\n",
    "bcftools index s47880_D2.GATK.indel.vcf.gz\n",
    "bcftools index s47880_D3.GATK.indel.vcf.gz\n",
    "bcftools isec -n +3 s47880_D1.GATK.indel.vcf.gz s47880_D2.GATK.indel.vcf.gz s47880_D3.GATK.indel.vcf.gz -p D123\n",
    "\n",
    "```"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
