{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Transfer data from Imstor to scratch space within O2 server**\n",
    "\n",
    "```bash\n",
    "rsync -aPr ajn16@transfer.rc.hms.harvard.edu:/n/files/ImStor/sorger/data/rnaseq/ajit_johns\n",
    "on/melanoma_rarecyte .\n",
    "```\n",
    "***\n",
    "\n",
    "**INTIAL QUALITY CONTROL**\n",
    "**FASTQC ANALYSIS**\n",
    "\n",
    "```bash\n",
    "mkdir fastqc\n",
    "cd fastqc\n",
    "vim fastqc.run\n",
    "```\n",
    "***\n",
    "\n",
    "```bash\n",
    "#!/bin/bash\n",
    "#SBATCH -p short                # partition name\n",
    "#SBATCH -t 0-2:00               # hours:minutes runlimit after which job will be killed\n",
    "#SBATCH -c 16                    # number of cores requested -- this needs to be greater than or equal to the number of cores you plan to use to run your job\n",
    "#SBATCH --job-name fastqc       # Job name\n",
    "#SBATCH --mem=70G\n",
    "#SBATCH -o %fastqc.out               # File to which standard out will be written\n",
    "#SBATCH -e %fastqc.err               # File to which standard err will be written\n",
    "#SBATCH --mail-type=END         # Type of email notification- BEGIN,END,FAIL,ALL\n",
    "#SBATCH --mail-user=ajitj_nirmal@dfci.harvard.edu   # Email to which notifications will be sent\n",
    "\n",
    "\n",
    "## Changing directories to where the fastq files are located\n",
    "cd /n/scratch2/ajit/melanoma_rarecyte/raw_files\n",
    "\n",
    "## Loading modules required for script commands\n",
    "module load fastqc/0.11.5\n",
    "\n",
    "## Running FASTQC\n",
    "fastqc -t 8 *.fastq\n",
    "\n",
    "## Move files \n",
    "mv *fastqc* ../fastqc\n",
    "```\n",
    "***\n",
    "\n",
    "**Run the job**\n",
    "\n",
    "```bash\n",
    "sbatch fastqc.run\n",
    "```\n",
    "\n",
    "**MultiQC analysis for aggregateing results across many samples into a single report**\n",
    "\n",
    "```bash\n",
    "cd round1/\n",
    "module load gcc/6.2.0  python/2.7.12 multiqc/1.5\n",
    "multiqc .\n",
    "```\n",
    "\n",
    "***\n",
    "\n",
    "**Trimming the reads using trimmomatic**\n",
    "\n",
    "```bash\n",
    "\n",
    "mkdir trimmomatic\n",
    "cp -r /n/scratch2/ajit/melanoma_rarecyte/raw_files/* /n/scratch2/ajit/melanoma_rarecyte/trimmomatic\n",
    "\n",
    "vim auto_trim.sh\n",
    "\n",
    "module load trimmomatic/0.36\n",
    "\n",
    "# Execution\n",
    "# Case(1) run on a couple of PE files with extension *.fq\n",
    "# $ sh auto_trim.sh *.fastq\n",
    "\n",
    "red=`tput setaf 1`\n",
    "green=`tput setaf 2`\n",
    "yellow=`tput setaf 3`\n",
    "reset=`tput sgr0`\n",
    "count=0\n",
    "\n",
    "usage ()\n",
    "{\n",
    "  echo -e \"${green}Usage: sh auto_trim [*.extension]\\n \\\n",
    "      extension: <fq> or <fastq> or <fq.gz> or <fastq.gz>\\n \\\n",
    "      example: sh auto_trim.sh *.fq.gz\\n ${reset}\\n\\\n",
    "${yellow}Help:  sh autotrim -h or --help${reset}\"\n",
    "  exit\n",
    "}\n",
    "\n",
    "file_not_found ()\n",
    "{\n",
    "echo -e \"\\n${red}FileNotFoundError: No such file with extension $@ found!${reset}\"\n",
    "echo -e \"${green}Supported extensions are: <.fq> or <.fastq> or <.fq.gz> or <.fastq.gz>${reset}\\n\"\n",
    "exit \n",
    "}\n",
    "\n",
    "file_name_error ()\n",
    "{\n",
    "echo -e \"\\n${red}Filename Error: Paired end file names should contain _R1 _R2${reset}\"\n",
    "echo -e \"${green}Example: test_R1.fq.gz, test_R2.fq.gz${reset}\\n\"\n",
    "exit \n",
    "}\n",
    "\n",
    "file_extension_error ()\n",
    "{\n",
    "echo -e \"\\n${red}FileExtensionError: Invalid extension${reset}\"\n",
    "echo -e \"${green}Supported extensions are: <.fq> or <.fastq> or <.fq.gz> or <.fastq.gz>${reset}\\n\"\n",
    "exit     \n",
    "}\n",
    "\n",
    "if [[ ( $1 == '-h' ) || ( $1 == '--help') ]] ;then\n",
    " usage\n",
    "elif [[ $# -eq 0 ]] ;then\n",
    " echo \"${red}Error: No parameter(s) provided${reset}\"\n",
    " usage\n",
    " exit 0\n",
    "else\n",
    "  for i in $@; do\n",
    "        count=$((count+1))\n",
    "        if [ -f $i ] ;then\n",
    "            if [[ (${i#*.} == \"fastq.gz\") || (${i#*.} == \"fq.gz\") || (${i#*.} == \"fastq\") || (${i#*.} == \"fq\") ]] ;then \n",
    "                if echo $1 | grep -q -e \"_R1\" -e \"_R2\"; then\n",
    "\t\t   if [[ $count%2 -ne 0 ]]; then\n",
    "                       sample_name=`echo $i | awk -F \"_R1\"  '{print $1}'`\n",
    "                       extension=`echo $i | awk -F \"_R1\"  '{print $2}'`\n",
    "                       R1=${sample_name}_R1${extension}\n",
    "\t               R2=${sample_name}_R2${extension}\n",
    "\t\t       R1_pair=${sample_name}_R1_pair${extension}\n",
    "\t\t       R1_unpair=${sample_name}_R1_unpair${extension}\n",
    "\t\t       R2_pair=${sample_name}_R2_pair${extension}\n",
    "\t\t       R2_unpair=${sample_name}_R2_unpair${extension}\n",
    "                       echo -e \"\\n${yellow}[Running trimmomatic for sample] ${sample_name} at `whoami`${reset}\\n\"\n",
    "                       date && time java -jar $TRIMMOMATIC/trimmomatic-0.36.jar \\\n",
    "                       PE \\\n",
    "                       $R1 $R2 $R1_pair $R1_unpair $R2_pair $R2_unpair \\\n",
    "                       HEADCROP:7 \\\n",
    "                       LEADING:20 \\\n",
    "                       TRAILING:20 \\\n",
    "                       SLIDINGWINDOW:20:20 \\\n",
    "                       MINLEN:50 \n",
    "                       rm *unpair*\n",
    "\n",
    "                   fi\n",
    "                else file_name_error\n",
    "                fi      \n",
    "            elif [[ (${i#*.} == \"sh\") || (${i#*.} == \"sh~\")  ]] ;then\n",
    "                 echo -n\n",
    "            else\n",
    "                echo -ne \"${red}Check:$i${reset}\"\n",
    "                file_extension_error  \n",
    "            fi     \n",
    "        else\n",
    "\t    file_not_found $@\n",
    "        fi\n",
    "  done\n",
    "fi\n",
    "```\n",
    "\n",
    "***\n",
    "\n",
    "**Run Trimmomatic**\n",
    "\n",
    "```bash\n",
    "sh auto_trim.sh *.fastq\n",
    "```\n",
    "***\n",
    "\n",
    "```bash\n",
    "cd trimmomatic\n",
    "find . -type f \\! -name \"*pair*\" -delete\n",
    "```\n",
    "```bash\n",
    "# Change the file names so that the old alignment.csv can be used\n",
    "for filename in *.fastq; do \n",
    "    [ -f \"$filename\" ] || continue\n",
    "    mv \"$filename\" \"${filename//_pair/}\"\n",
    "done\n",
    "```\n",
    "\n",
    "**Run Aignment**\n",
    "\n",
    "```bash\n",
    "mkdir reference\n",
    "wget ftp://ftp.ensembl.org/pub/release-96/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh38.cdna.all.fa.gz\n",
    "wget ftp://ftp.ensembl.org/pub/release-96/gtf/homo_sapiens/Homo_sapiens.GRCh38.96.gtf.gz\n",
    "gunzip *\n",
    "```\n",
    "\n",
    "***\n",
    "**Prepare the configuration/yaml file**\n",
    "\n",
    "```bash\n",
    "wget https://labsyspharm.github.io/rnaseq/example_settings/rna_seq/O2.yaml\n",
    "vim O2.yaml\n",
    "\n",
    "details:\n",
    "  - analysis: RNA-seq\n",
    "    genome_build: hg38\n",
    "    algorithm:\n",
    "      transcriptome_fasta: /n/scratch2/ajit/melanoma_rarecyte/reference/Homo_sapiens.GRCh38.cdna.all.fa\n",
    "      transcriptome_gtf: /n/scratch2/ajit/melanoma_rarecyte/reference/Homo_sapiens.GRCh38.96.gtf\n",
    "      aligner: hisat2\n",
    "      strandedness: unstranded\n",
    "upload:\n",
    "  dir: ../final\n",
    "```\n",
    "\n",
    "***\n",
    "**Create a sample description file**\n",
    "\n",
    "```bash\n",
    "cd trimmomatic\n",
    "(echo 'samplename,description'; for f in trimmomatic/*fastq*; do readlink -f $f | perl -pe 's/(.*?_(S[0-9]+)_.*)/\\1,\\2/'; done) > alignment.csv\n",
    "```\n",
    "***\n",
    "\n",
    "**Intiate bcBio**\n",
    "\n",
    "```bash\n",
    "module load bcbio/latest\n",
    "unset PYTHONPATH\n",
    "bcbio_nextgen.py -w template O2.yaml alignment.csv trimmomatic/\n",
    "```\n",
    "***\n",
    "\n",
    "**Run bcBio**\n",
    "```bash\n",
    "vim submit_bcbio.sh\n",
    "```\n",
    "\n",
    "```bash\n",
    "#!/bin/sh\n",
    "#SBATCH -p medium\n",
    "#SBATCH -J bcbio_O2\n",
    "#SBATCH -c 16                \n",
    "#SBATCH -o run.o\n",
    "#SBATCH -e run.e\n",
    "#SBATCH -t 1-00:00\n",
    "#SBATCH --cpus-per-task=3\n",
    "#SBATCH --mem=64G\n",
    "#SBATCH --mail-type=END         # Type of email notification- BEGIN,END,FAIL,ALL\n",
    "#SBATCH --mail-user=ajitj_nirmal@dfci.harvard.edu   # Email to which notifications will be sent\n",
    "\n",
    "export PATH=/n/app/bcbio/tools/bin:$PATH\n",
    "bcbio_nextgen.py ../config/alignment.yaml \\\n",
    "    -n 24 -t ipython -s slurm -q medium -r t=1-00:00 --timeout 2000\n",
    "    \n",
    "sbatch submit_bcbio.sh\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
