{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Step 1: Rename files and make sure every chip file has an associated input file\n",
    "All the data was manually reviewed and renamed for human readability. Since there was a single input sample seqenced that would serve as an input for several chipseq files, the input was duplicated such that each chipseq file has an associated input file. The data was then transfered to O2 for processing. <br>\n",
    "\n",
    "```\n",
    "srun --pty -p interactive --mem 80G -t 0-06:00 /bin/bash\n",
    "gunzip *.gz\n",
    "```\n",
    "\n",
    "#### Step 2: Create a sample description file and edit to include metadata regarding the samples\n",
    "\n",
    "```\n",
    "(echo 'samplename,description'; for f in raw_files/*fq*; do readlink -f $f | perl -pe 's/(.*?_(S[0-9]+)_.*)/\\1,\\2/'; done) > alignment.csv\n",
    "```\n",
    "\n",
    "Edit the alignment.csv file to include, description, batch and phenotype. <br>\n",
    "\n",
    "#### Step 3: Create a Yaml File\n",
    "\n",
    "```\n",
    "details:\n",
    "- analysis: chip-seq\n",
    "  genome_build: hg19\n",
    "  algorithm:\n",
    "    aligner: bowtie2\n",
    "    peakcaller: [macs2]\n",
    "  resources:\n",
    "    macs2:\n",
    "      options: [\"-q 0.05\"]\n",
    "```\n",
    "\n",
    "##### Step 4: Intiate bcBio.\n",
    "\n",
    "```\n",
    "module load bcbio/latest\n",
    "unset PYTHONPATH\n",
    "bcbio_nextgen.py -w template O2.yaml alignment.csv raw_files/\n",
    "```\n",
    "\n",
    "##### Step-5: Create Submission script to O2.\n",
    "\n",
    "```\n",
    "vim submit_bcbio.sh\n",
    "\n",
    "#!/bin/sh\n",
    "#SBATCH -p priority\n",
    "#SBATCH -J wenchao\n",
    "#SBATCH -o run.o\n",
    "#SBATCH -e run.e\n",
    "#SBATCH -t 7-00:00\n",
    "#SBATCH --cpus-per-task=20\n",
    "#SBATCH --mem=120G\n",
    "#SBATCH --mail-type=END         # Type of email notification- BEGIN,END,FAIL,ALL\n",
    "#SBATCH --mail-user=ajitj_nirmal@dfci.harvard.edu   # Email to which notifications will be sent\n",
    "\n",
    "export PATH=/n/app/bcbio/tools/bin:$PATH\n",
    "bcbio_nextgen.py ../config/alignment.yaml \\\n",
    "    -n 24 -t local\n",
    "```\n",
    "\n",
    "##### Step-6: Submit job to O2 for processing\n",
    "\n",
    "```\n",
    "cp submit_bcbio.sh alignment/work\n",
    "cd alignment/work\n",
    "sbatch submit_bcbio.sh\n",
    "```\n",
    "\n",
    "#### Continue Analysis in R to annotate the peaks\n",
    "\n",
    "```\n",
    "#WD\n",
    "setwd (\"C:/Users/ajn16/Dropbox (Partners HealthCare)/Data/wenchao_chipseq/result bed files\")\n",
    "\n",
    "# Load Lib\n",
    "library(ChIPpeakAnno)\n",
    "library(org.Hs.eg.db)\n",
    "library(GenomicFeatures)\n",
    "library(TxDb.Hsapiens.UCSC.hg19.knownGene)\n",
    "data(TSS.human.GRCh37)\n",
    "\n",
    "# Function to annotate peaks with hg19 genome\n",
    "chipseq_processing <- function (bedfile){\n",
    "  data <- toGRanges(bedfile, format=\"BED\", header=FALSE)\n",
    "  anno <- annotatePeakInBatch(data, AnnotationData=TSS.human.GRCh37)\n",
    "  anno <- addGeneIDs(annotatedPeak=anno, orgAnn=\"org.Hs.eg.db\", IDs2Add=\"symbol\")\n",
    "  ucsc.hg19.knownGene <- genes(TxDb.Hsapiens.UCSC.hg19.knownGene)\n",
    "  anno <- annotatePeakInBatch(data, AnnotationData=ucsc.hg19.knownGene)\n",
    "  anno <- addGeneIDs(annotatedPeak=anno, orgAnn=\"org.Hs.eg.db\", feature_id_type=\"entrez_id\",IDs2Add=\"symbol\")\n",
    "  write.csv(anno, file = paste(\"peak files/\", bedfile, \".csv\"))\n",
    "}\n",
    "\n",
    "\n",
    "# Processing\n",
    "chipseq_processing (bedfile = \"H3K4me3C1_summits.bed\")\n",
    "chipseq_processing (bedfile = \"H3K4me3C2_summits.bed\")\n",
    "chipseq_processing (bedfile = \"H3K4me3L_summits.bed\")\n",
    "chipseq_processing (bedfile = \"H3K27ACC1_summits.bed\")\n",
    "chipseq_processing (bedfile = \"H3K27ACC2_summits.bed\")\n",
    "chipseq_processing (bedfile = \"H3K27ACL_summits.bed\")\n",
    "chipseq_processing (bedfile = \"H3K27me3C1_summits.bed\")\n",
    "chipseq_processing (bedfile = \"H3K27me3C2_summits.bed\")\n",
    "chipseq_processing (bedfile = \"H3K27me3L_summits.bed\")\n",
    "chipseq_processing (bedfile = \"IKZF1A_summits.bed\")\n",
    "chipseq_processing (bedfile = \"IKZF1B_summits.bed\")\n",
    "chipseq_processing (bedfile = \"IKZF1C_summits.bed\")\n",
    "chipseq_processing (bedfile = \"IKZF1C1_summits.bed\")\n",
    "chipseq_processing (bedfile = \"IKZF1C2_summits.bed\")\n",
    "chipseq_processing (bedfile = \"IKZF1C3_summits.bed\")\n",
    "chipseq_processing (bedfile = \"K4me3A_summits.bed\")\n",
    "chipseq_processing (bedfile = \"K4me3B_summits.bed\")\n",
    "chipseq_processing (bedfile = \"K4me3C_summits.bed\")\n",
    "chipseq_processing (bedfile = \"K27AcA_summits.bed\")\n",
    "chipseq_processing (bedfile = \"K27AcB_summits.bed\")\n",
    "chipseq_processing (bedfile = \"K27AcC_summits.bed\")\n",
    "chipseq_processing (bedfile = \"ZFP91A_summits.bed\")\n",
    "chipseq_processing (bedfile = \"ZFP91B_summits.bed\")\n",
    "chipseq_processing (bedfile = \"ZFP91C_summits.bed\")\n",
    "chipseq_processing (bedfile = \"ZFP91C1_summits.bed\")\n",
    "chipseq_processing (bedfile = \"ZFP91C2_summits.bed\")\n",
    "chipseq_processing (bedfile = \"ZFP91C3_summits.bed\")\n",
    "```\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Manual chipseq run for an example file\n",
    "\n",
    "**Step 1:**\n",
    "\n",
    "```\n",
    "mkdir reference\n",
    "wget ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/hg19.zip\n",
    "unzip hg19.zip\n",
    "```\n",
    "\n",
    "**Align to the reference genome**\n",
    "\n",
    "```\n",
    "#!/bin/sh\n",
    "#SBATCH -p priority\n",
    "#SBATCH -J wen_alignment\n",
    "#SBATCH -o run.o\n",
    "#SBATCH -e run.e\n",
    "#SBATCH -t 1-00:00\n",
    "#SBATCH --cpus-per-task=20\n",
    "#SBATCH --mem=100G\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH --mail-user=ajit_nirmal@hms.harvard.edu\n",
    "\n",
    "module load gcc/6.2.0 bowtie2/2.3.4.3\n",
    "/n/app/bowtie2/2.3.4.3/bin/bowtie2 -x hg19 -1 /n/scratch2/ajit/wenchao/manual_run/raw_files/IKZF1_C1_S255_L001_R1_001.fastq.gz -2 /n/scratch2/ajit/wenchao/manual_run/raw_files/IKZF1_C1_S255_L001_R2_001.fastq.gz -S IKZF1.sam\n",
    "```\n",
    "\n",
    "**Run alignment**\n",
    "\n",
    "```\n",
    "mv alignment.sh reference/\n",
    "sbatch alignment.sh\n",
    "```\n",
    "\n",
    "**Convert SAM to BAM files**\n",
    "\n",
    "```\n",
    "cd reference/\n",
    "mv IKZF1.sam ../bam\n",
    "\n",
    "#!/bin/sh\n",
    "#SBATCH -p priority\n",
    "#SBATCH -J wen_samtobam\n",
    "#SBATCH -o run.osamtobam\n",
    "#SBATCH -e run.esamtobam\n",
    "#SBATCH -t 0-06:00\n",
    "#SBATCH --cpus-per-task=20\n",
    "#SBATCH --mem=100G\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH --mail-user=ajit_nirmal@hms.harvard.edu\n",
    "\n",
    "module load gcc/6.2.0 samtools/1.9\n",
    "samtools view -S -b IKZF1.sam > IKZF1.bam\n",
    "samtools sort IKZF1.bam -o sorted_IKZF1.bam\n",
    "```\n",
    "\n",
    "**Peak Calling**\n",
    "\n",
    "```\n",
    "\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
